packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,358 @@
+/// If the maximum width given to the picker is greater than [_kPickerWidth],
+/// the leftmost and rightmost column will be extended equally so that the
+/// widths match, and the picker is in the center.
+///
+/// If the maximum width given to the picker is smaller than [_kPickerWidth],
+/// the picker's layout will be broken.
+
+
+/// A countdown timer picker in iOS style.
+///
+/// This picker shows duration as hour, minute and second spinners. The duration
+/// showed has to be non negative and is limited to 23 hours 59 minutes 59 seconds.
+///
+/// Example: [12 hours |  3 min |  0 sec]
+class CupertinoCountdownTimerPicker extends StatefulWidget {
+  /// Constructs an iOS style countdown timer picker.
+  ///
+  /// [onTimerDurationChanged] is the callback when the selected duration changes
+  /// and must not be null.
+  ///
+  /// [initialTimerDuration] defaults to 0 second and is limited from 0 second
+  /// to 23 hours 59 minutes 59 seconds. Only hour, minute, and second values
+  /// are extracted from [initialTimerDuration], so specifying other fields like
+  /// millisecond or microsecond will not affect anything.
+  ///
+  /// [minuteInterval] is the granularity of the minute spinner. Must be a factor
+  /// of 60.
+  ///
+  /// [secondInterval] is the granularity of the second spinner. Must be a factor
+  /// of 60.
+  CupertinoCountdownTimerPicker({
+    this.initialTimerDuration = const Duration(),
+    this.minuteInterval = 1,
+    this.secondInterval = 1,
+    @required this.onTimerDurationChanged,
+  }) : assert(onTimerDurationChanged != null),
+       assert(initialTimerDuration >= const Duration(seconds: 0)),
+       assert(initialTimerDuration < const Duration(days: 1)),
+       assert(minuteInterval > 0 && 60 % minuteInterval == 0),
+       assert(secondInterval > 0 && 60 % secondInterval == 0),
+       assert(initialTimerDuration.inMinutes % minuteInterval == 0),
+       assert(initialTimerDuration.inSeconds % secondInterval == 0);
+
+  /// The initial duration of the countdown timer.
+  final Duration initialTimerDuration;
+
+  /// The granularity of the minute spinner. Must be a factor of 60.
+  final int minuteInterval;
+
+  /// The granularity of the second spinner. Must be a factor of 60.
+  final int secondInterval;
+
+  /// Callback when the timer duration changes.
+  final ValueChanged<Duration> onTimerDurationChanged;
+
+  @override
+  State<StatefulWidget> createState() => _CountdownTimerState();
+}
+
+class _CountdownTimerState extends State<CupertinoCountdownTimerPicker> {
+  // The currently selected values of the picker.
+  int selectedHour;
+  int selectedMinute;
+  int selectedSecond;
+
+  // Controllers for the 3 units: hour, minute, and second.
+  FixedExtentScrollController hourController;
+  FixedExtentScrollController minuteController;
+  FixedExtentScrollController secondController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    selectedHour = widget.initialTimerDuration.inHours;
```

Comment:
What happens when the user rebuilds the widget? i.e. didUpdateWidget

Reply:
The picker will still have its current duration (will not be reset to initial duration).

Reply:
I'm not sure what should actually be updated in didUpdateWidget, so if you have any suggestion please let me know.